src/bscPlugin/codeActions/XmlScopeCodeActionsProcessor.ts

```diff
@@ -0,0 +1,58 @@
+import type { CodeAction, Range } from 'vscode-languageserver';
+import { isBrsFile } from '../../astUtils/reflection';
+import { DiagnosticCodeMap } from '../../DiagnosticMessages';
+import type { BscFile, BsDiagnostic } from '../../interfaces';
+import util from '../../util';
+import type { XmlScope } from '../../XmlScope';
+
+// eslint-disable-next-line @typescript-eslint/no-extraneous-class
+export class XmlScopeCodeActionProcessor {
+    public constructor(
+        public scope: XmlScope,
+        public file: BscFile,
+        public range: Range,
+        public diagnostics: BsDiagnostic[],
+        public codeActions: CodeAction[]
+    ) {
+
+    }
+
+    public process() {
+        for (const diagnostic of this.diagnostics) {
+            if (diagnostic.code === DiagnosticCodeMap.callToUnknownFunction) {
+                this.importXmlScript(diagnostic);
+            }
+        }
+    }
+
+    public importXmlScript(diagnostic: BsDiagnostic) {
+        //functionName is stored on this specific diagnostic
+        const lowerFunctionName = (diagnostic as any).functionName.toLowerCase();
+
+        //find every file with this function defined
+        for (const key in this.scope.program.files) {
+            const file = this.scope.program.files[key];
+            if (isBrsFile(file)) {
+                //TODO handle namespace-relative function calls
+                const stmt = file.parser.references.functionStatementLookup.get(lowerFunctionName);
+                const slashOpenToken = this.scope.xmlFile.parser.ast.component?.ast.SLASH_OPEN?.[0];
+                if (stmt && slashOpenToken) {
+                    const pkgPath = util.getRokuPkgPath(file.pkgPath);
+                    this.codeActions.push(
+                        util.createCodeAction({
+                            title: `Import "${pkgPath}" into component "${this.scope.xmlFile.componentName.text ?? this.scope.name}"`,
+                            // diagnostics: [diagnostic]
+                            changes: [{
+                                filePath: this.scope.xmlFile.pathAbsolute,
+                                newText: `  <script type="text/brightscript" uri="${pkgPath}" />\n`,
```

Comment:
Does it honor the file's indentation? Surely that should be possible to get he IDE do automatically.
Would be worth having an `insertLine` action where you only provide the line number.

Reply:
No, it doesn't currently honor the file's indentation. I agree, this needs some more work. I'll work on it.

Reply:
I'll handle this in a separate PR, since this code is no longer included in this PR